### PR TITLE
tests: enable tests for generators 

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
tests

**Did you add tests for your changes?**
yes
**If relevant, did you update the documentation?**
NA
**Summary**

Enable skipped tests. 

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

**Review by evilebottnawi — DISMISSED** [review]

File: test/env/object/object-env.test.js [hunk 1]
```diff
@@ -7,15 +7,15 @@ const { sync: spawnSync } = execa;
 const { run } = require('../../utils/test-utils');
 
 describe('env object', () => {
-    it('is able to set env for an object', () => {
-        const { stderr, stdout, exitCode } = run(__dirname);
+    it('is able to set env for an object', async () => {
+        const { stderr, stdout, exitCode } = await run(__dirname);
```
Comment (anshumanv) [inline test/env/object/object-env.test.js:11]: run is synchronous right?

File: test/env/object/object-env.test.js [hunk 2]
```diff
@@ -7,15 +7,15 @@ const { sync: spawnSync } = execa;
...
 
         expect(exitCode).toBe(0);
         expect(stderr).toBeFalsy();
         expect(stdout).toBeTruthy();
 
-        const executable = path.join(__dirname, './bin/main.js');
-        const bundledScript = spawnSync('node', [executable]);
+        const executable = await path.join(__dirname, './bin/main.js');
```
Comment (evilebottnawi) [inline test/env/object/object-env.test.js:17]: path.join is not async :smile:

File: test/env/object/object-env.test.js (hunk 1 above)
Comment (snitin315) [reply test/env/object/object-env.test.js:11]: yes updating.

File: test/env/array/array-env.test.js [hunk 3]
```diff
@@ -10,15 +10,15 @@ const devFile = path.join(__dirname, './bin/dev.js');
 const prodFile = path.join(__dirname, './bin/prod.js');
 
 describe('env array', () => {
-    it('is able to set two different environments for an array configuration', () => {
-        const { stderr, stdout, exitCode } = run(__dirname);
+    it('is able to set two different environments for an array configuration', async () => {
+        const { stderr, stdout, exitCode } = await run(__dirname);
```
Comment (snitin315) [inline test/env/array/array-env.test.js:14]:
```suggestion
        const { stderr, stdout, exitCode } = run(__dirname);
```

File: test/env/object/object-env.test.js (hunk 1 above)
Comment (snitin315) [reply test/env/object/object-env.test.js:11]:
```suggestion
        const { stderr, stdout, exitCode } = run(__dirname);
```

File: test/env/object/object-env.test.js (hunk 2 above)
Comment (snitin315) [reply test/env/object/object-env.test.js:17]:
```suggestion
        const executable = path.join(__dirname, './bin/main.js');
```

Comment (snitin315) [inline test/env/object/object-env.test.js:17]:
```suggestion
        const executable = path.join(__dirname, './bin/main.js');
```

**Review by evilebottnawi — DISMISSED** [review]

File: test/env/array/array-env.test.js [hunk 4]
```diff
@@ -10,15 +10,15 @@ const devFile = path.join(__dirname, './bin/dev.js');
...
+    it('is able to set two different environments for an array configuration', async () => {
         const { stderr, stdout, exitCode } = run(__dirname);
 
         expect(exitCode).toBe(0);
         expect(stderr).toBeFalsy();
         expect(stdout).toBeTruthy();
 
-        const devScript = spawnSync('node', [devFile]);
-        const prodScript = spawnSync('node', [prodFile]);
+        const devScript = await spawnSync('node', [devFile]);
```
Comment (anshumanv) [inline test/env/array/array-env.test.js:20]: why need await when spawnSync is synchronous by nature?

File: test/env/array/array-env.test.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ const devFile = path.join(__dirname, './bin/dev.js');
 const prodFile = path.join(__dirname, './bin/prod.js');
 
 describe('env array', () => {
-    it('is able to set two different environments for an array configuration', () => {
+    it('is able to set two different environments for an array configuration', async () => {
```
Comment (snitin315) [inline test/env/array/array-env.test.js:13]:
```suggestion
    it('is able to set two different environments for an array configuration', () => {
```

File: test/env/object/object-env.test.js [hunk 6]
```diff
@@ -7,7 +7,7 @@ const { sync: spawnSync } = execa;
 const { run } = require('../../utils/test-utils');
 
 describe('env object', () => {
-    it('is able to set env for an object', () => {
+    it('is able to set env for an object', async () => {
```
Comment (snitin315) [inline test/env/object/object-env.test.js:10]:
```suggestion
    it('is able to set env for an object', () => {
```

File: test/env/array/array-env.test.js (hunk 4 above)
Comment (snitin315) [reply test/env/array/array-env.test.js]: I have reverted it lets see if CI is happy with it 😄

**Review by evilebottnawi — APPROVED** [review]

**Review by jamesgeorge007 — APPROVED** [review]

**Pull request merged**